dfget/core/downloader/p2p_downloader/client_stream_writer.go

```diff
@@ -68,11 +71,12 @@ type ClientStreamWriter struct {
 }
 
 // NewClientStreamWriter creates and initialize a ClientStreamWriter instance.
-func NewClientStreamWriter(clientQueue queue.Queue, api api.SupernodeAPI, cfg *config.Config) *ClientStreamWriter {
+func NewClientStreamWriter(clientQueue, notifyQueue queue.Queue, api api.SupernodeAPI, cfg *config.Config) *ClientStreamWriter {
```

Comment:
I think we should abstract an interface for client writer, WDYT?

Reply:
agreed, there is an ambitious plan to do this, but not in this pr.

Reply:
OK